pkg/api/v20220201/openshiftcluster.go

```diff
@@ -0,0 +1,277 @@
+	// The client ID used for the cluster.
+	ClientID string `json:"clientId,omitempty" mutable:"true"`
+
+	// The client secret used for the cluster.
+	ClientSecret string `json:"clientSecret,omitempty" mutable:"true"`
+}
+
+// SoftwareDefinedNetwork constants.
+type SoftwareDefinedNetwork string
+
+const (
+	SoftwareDefinedNetworkOVNKubernetes SoftwareDefinedNetwork = "OVNKubernetes"
+	SoftwareDefinedNetworkOpenShiftSDN  SoftwareDefinedNetwork = "OpenShiftSDN"
+)
+
+// NetworkProfile represents a network profile.
+type NetworkProfile struct {
+	// The software defined network (SDN) to use when installing the cluster.
+	SoftwareDefinedNetwork SoftwareDefinedNetwork `json:"softwareDefinedNetwork,omitempty"`
+
+	// The CIDR used for OpenShift/Kubernetes Pods.
+	PodCIDR string `json:"podCidr,omitempty"`
+
+	// The CIDR used for OpenShift/Kubernetes Services.
+	ServiceCIDR string `json:"serviceCidr,omitempty"`
+}
+
+// EncryptionAtHost represents encryption at host state
+type EncryptionAtHost string
+
+// EncryptionAtHost constants
+const (
+	EncryptionAtHostEnabled  EncryptionAtHost = "Enabled"
+	EncryptionAtHostDisabled EncryptionAtHost = "Disabled"
+)
+
+// MasterProfile represents a master profile.
+type MasterProfile struct {
+	// The size of the master VMs.
+	VMSize VMSize `json:"vmSize,omitempty"`
+
+	// The Azure resource ID of the master subnet.
+	SubnetID string `json:"subnetId,omitempty"`
+
+	// Whether master virtual machines are encrypted at host.
+	EncryptionAtHost EncryptionAtHost `json:"encryptionAtHost,omitempty"`
+
+	// The resource ID of an associated DiskEncryptionSet, if applicable.
+	DiskEncryptionSetID string `json:"diskEncryptionSetId,omitempty"`
+}
+
+// VMSize represents a VM size.
+type VMSize string
+
+// VMSize constants
+const (
+	VMSizeStandardD2sV3 VMSize = "Standard_D2s_v3"
+
+	VMSizeStandardD4asV4  VMSize = "Standard_D4as_v4"
+	VMSizeStandardD8asV4  VMSize = "Standard_D8as_v4"
+	VMSizeStandardD16asV4 VMSize = "Standard_D16as_v4"
+	VMSizeStandardD32asV4 VMSize = "Standard_D32as_v4"
+
+	VMSizeStandardD4sV3  VMSize = "Standard_D4s_v3"
+	VMSizeStandardD8sV3  VMSize = "Standard_D8s_v3"
+	VMSizeStandardD16sV3 VMSize = "Standard_D16s_v3"
+	VMSizeStandardD32sV3 VMSize = "Standard_D32s_v3"
+
+	VMSizeStandardE4sV3   VMSize = "Standard_E4s_v3"
+	VMSizeStandardE8sV3   VMSize = "Standard_E8s_v3"
+	VMSizeStandardE16sV3  VMSize = "Standard_E16s_v3"
+	VMSizeStandardE32sV3  VMSize = "Standard_E32s_v3"
+	VMSizeStandardE64isV3 VMSize = "Standard_E64is_v3"
+	VMSizeStandardE64iV3  VMSize = "Standard_E64i_v3"
+
+	VMSizeStandardF4sV2  VMSize = "Standard_F4s_v2"
+	VMSizeStandardF8sV2  VMSize = "Standard_F8s_v2"
+	VMSizeStandardF16sV2 VMSize = "Standard_F16s_v2"
+	VMSizeStandardF32sV2 VMSize = "Standard_F32s_v2"
+	VMSizeStandardF72sV2 VMSize = "Standard_F72s_v2"
+
+	VMSizeStandardM128ms VMSize = "Standard_M128ms"
+	VMSizeStandardG5     VMSize = "Standard_G5"
+	VMSizeStandardGS5    VMSize = "Standard_GS5"
```

Comment:
Are we OK to get the SKUs in https://github.com/Azure/ARO-RP/pull/1751/files added here and other places?

That draft PR L-series is mostly a placeholder - we should be good to close it when this goes out with the new SKUs (https://github.com/Azure/ARO-RP/pull/1790 is a prerequisite).